### ISSUE TITLE
Unused variable lint causes compiler error

### ISSUE THREAD

**Issue by 0XPW87** [rust-lang/rust-clippy#16657]
### Summary

Cargo clippy fix attempts to rename an unused_variable to _unused_variable, however if unreachable code exists downstream, the fix is not applied there, causing a compiler error and the fixes to not be applied.

Clippy sent me here, although I can imagine this is a rather niche issue...

### Reproducer

Code:

```rust
match(v){
    ...,
    EnumVar::Var2(unused_var) => {
        //  -------- `_unused_var` defined here
        todo!()
        Some(unused_var) //<-- error occurrs here, 'cannot find value `unused_var` in this scope',
    }
}
```

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by profetia** [comment]
`unused_variables` is a rustc lint, not a clippy lint. You should consider open this issue in the rustc repo instead.

**Comment by 0XPW87** [comment]
Well, cargo clippy gave me this url to report to, which I did. Feel free to close.

**Comment by y21** [comment]
This is another instance of rust-lang/rust#107734, where the lint doesn't realize a variable is used after unreachable code. Closing as a duplicate.
